wireguard-broker/src/api/msgs.rs

```diff
@@ -17,25 +25,43 @@ pub struct Envelope<M: AsBytes + FromBytes> {
     pub payload: M,
 }
 
+/// Message format for requests to set a pre-shared key.
+/// # Example
+///
 #[repr(packed)]
 #[derive(AsBytes, FromBytes, FromZeroes)]
 pub struct SetPskRequest {
+    /// The pre-shared key.
     pub psk: [u8; 32],
+    /// The identifier of the peer.
     pub peer_id: [u8; 32],
+    /// The size for the interface
     pub iface_size: u8, // TODO: We should have variable length strings in lenses
+    /// The buffer for the interface.
     pub iface_buf: [u8; 255],
 }
 
 impl SetPskRequest {
+    /// Gets the interface specification as byte slice.
     pub fn iface_bin(&self) -> &[u8] {
         let len = self.iface_size as usize;
         &self.iface_buf[..len]
     }
 
+    /// Gets the interface specification as a `&str`.
+    ///
+    /// # Errors
+    /// Returns a [Utf8Error] if the interface specification isn't utf8 encoded.
     pub fn iface(&self) -> Result<&str, Utf8Error> {
         from_utf8(self.iface_bin())
     }
 
+    /// Sets the interface specification to `iface`. No check is made wether `icace` is correctly
```

Comment:
Typo icace

Reply:
Also: `whether`

Reply:
Seemingly wether is an English word with the same meaning as whether :)